ethcore/src/state.rs

```diff
@@ -410,10 +410,27 @@ impl fmt::Debug for State {
 
 impl Clone for State {
 	fn clone(&self) -> State {
+		let cache = {
+			let mut cache = HashMap::new();
+			for (key, val) in self.cache.borrow().iter() {
+				let key = key.clone();
+				match *val {
+					Some(ref acc) if acc.is_dirty() => {
+						cache.insert(key, Some(acc.clone()));
+					},
+					None => {
+						cache.insert(key, None);
```

Comment:
Doesn't a cached value of `None` indicate that the account was attempted to be cached but couldn't be found? See the implementation of `ensure_cached` -- `None` will only be assigned when the trie lookup returns nothing.

Probably better just not to insert at all for clean accounts.